### PR TITLE
feat: support shared secret

### LINKED ISSUE #124: Support shared secret for Admin API
See https://github.com/adobe/helix-admin/issues/282
> for some projects a shared secret based auth may be a good way to allow and revoke access to the admin API for a project via a shared secret that is sent from sidekick et al.

### PR DESCRIPTION
**Pull request by rofe**
Fix #124

### CODE REVIEW

**Review by kptdobe — APPROVED** [review]

File: src/sidekick/module.js [hunk 1]
```diff
@@ -417,7 +418,7 @@
   /**
    * Returns the share URL for the sidekick bookmarklet.
    * @private
-   * @param {Object} config The sidekick configuration
+   * @param {sidekickConfig} config The sidekick configuration
```
Comment (tripodsan) [inline src/sidekick/module.js:421]:
```suggestion
   * @param {SidekickConfig} config The sidekick configuration
```

File: src/sidekick/module.js [hunk 2]
```diff
@@ -520,22 +522,49 @@
...
 
   /**
-   * Creates an Admin URL for an API and path.
+   * Makes an Admin API call.
    * @private
-   * @param {Object} config The sidekick configuration
-   * @param {string} api The API endpoint to call
-   * @param {string} path The current path
-   * @returns {string} The admin URL
+   * @param {sidekickConfig} config The sidekick configuration
```
Comment (tripodsan) [inline src/sidekick/module.js:527]:
```suggestion
   * @param {SidekickConfig} config The sidekick configuration
```

File: src/sidekick/module.js [hunk 3]
```diff
@@ -520,22 +522,49 @@
...
       path,
     ].join(''));
+    if (editUrl) {
+      apiUrl.search = new URLSearchParams([
+        ['editUrl', editUrl],
+      ]).toString();
+    }
+    const headers = {};
+    if (token) {
+      headers['x-token'] = token;
```
Comment (tripodsan) [inline src/sidekick/module.js:561]:
```suggestion
      headers['x-preauth-token'] = token;
```

File: src/sidekick/module.js [hunk 4]
```diff
@@ -982,6 +1004,7 @@
      */
     loadContext(cfg) {
       this.config = initConfig(cfg);
+      console.log(this.config);
```
Comment (tripodsan) [inline src/sidekick/module.js:1007]:
```suggestion
      // console.log(this.config);
```

File: src/sidekick/module.js [hunk 5]
```diff
@@ -1405,8 +1428,11 @@
       try {
         if (config.hlx3) {
           // update preview
-          resp = await fetch(getAdminUrl(config, this.isContent() ? 'preview' : 'code', path),
-            { method: 'POST' });
+          resp = await callAdmin(config, {
+            api: this.isContent() ? 'preview' : 'code',
```
Comment (tripodsan) [inline src/sidekick/module.js:1432]: why not the `/status` api ?

**Review by tripodsan — CHANGES_REQUESTED** [review]
rename the token header, otherwise ok.

File: src/sidekick/module.js (hunk 1 above)
Comment (rofe) [reply src/sidekick/module.js:421]: This is a reference to [`sidekickConfig`](https://github.com/adobe/helix-sidekick/pull/125/files/a970ff06e1752e5dd241e1978dbb6fb8d4dc34ab#diff-2ba796749b8efb595287333808bcf9dc6b7fcaededc33a214038c961dda8c1ccR59) which is a virtual object that only exists in jsdoc...

Comment (tripodsan) [reply src/sidekick/module.js:421]: yes. but all class like structures and types should start upper case.

**Review by tripodsan — APPROVED** [review]

**Review by tripodsan — CHANGES_REQUESTED** [review]
requirements changed....

**Pull request closed without merging**